Services/Tagging/classes/class.ilTaggingSlateContentMenuGUI.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+/* Copyright (c) 1998-2019 ILIAS open source, Extended GPL, see docs/LICENSE */
+
+use ILIAS\Data\URI;
+
+/**
+ * Tagging slate UI
+ *
+ * @author killing@leifos.de
+ */
+class ilTaggingSlateContentMenuGUI extends ilTaggingSlateContentGUI
+{
+    /**
+     * show resources
+     *
+     * @param
+     * @return string
+     */
+    protected function showResourcesForTag()
+    {
+        // back button
+        $tag_cmd = $this->ctrl->getLinkTarget($this, "showTagCloud", "", true, false);
+        $items[] = $this->ui->factory()->button()->bulky(
+            $this->ui->factory()->symbol()->glyph()->back(),
+            $this->lng->txt("back"), "#")->withOnLoadCode(function ($id) use ($tag_cmd) {
+            return "$(\"#$id\").click(function() { 
+                il.Util.ajaxReplaceInner('$tag_cmd', 'il-tag-slate-container'); return false;
+            });";
+        });
+
+        $items[] = $this->ui->factory()->item()->standard(sprintf(
+            $this->lng->txt("tagging_resources_for_tag"),
+            "<i>" . $_GET["tag"] . "</i>"
```

Comment:
Please check if this is vulnerable against any XSS attacks. If so, please use `ilUtil::stripSlashes()`.

Reply:
Thx, this was indeed not XSS safe.
Within fixing this i encountered that the taggin has problems with quotings in general (Tags with quote cant be resolved in the Menu). This fix does not fix this problems completly (tag will now be resolved but always with an empty result) but solves at least the security issues.